exercises/matrix/src/test/java/MatrixTest.java

```diff
@@ -0,0 +1,184 @@
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Enclosed.class)
+public class MatrixTest {
+
+    @RunWith(Parameterized.class)
+    public static class GetFirstRowTest {
+        private Matrix matrix;
+        private int[] firstRow;
+
+        @Parameterized.Parameters(name = "{index}: checking first row of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0, 1}},
+                    {"0\\n1", new int[] {0}},
+                    {"0 1\\n2 3", new int[] {0, 1}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {0, 1, 2}}
+            });
+        }
+
+        public GetFirstRowTest(String matrixAsString, int[] firstRow) {
+            this.matrix = new Matrix(matrixAsString.replaceAll("\\\\n", "\n"));
+            this.firstRow = firstRow;
+        }
+
+        @Test
+        public void getFirstRowTest() {
+            assertArrayEquals(matrix.getRow(0), firstRow);
+        }
+    }
+
+    @RunWith(Parameterized.class)
+    public static class GetLastRowTest {
+        private Matrix matrix;
+        private int[] lastRow;
+
+        @Parameterized.Parameters(name = "{index}: checking last row of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0, 1}},
+                    {"0\\n1", new int[] {1}},
+                    {"0 1\\n2 3", new int[] {2, 3}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {6, 7, 8}}
+            });
+        }
+
+        public GetLastRowTest(String matrixAsString, int[] lastRow) {
+            this.matrix = new Matrix(matrixAsString.replaceAll("\\\\n", "\n"));
+            this.lastRow = lastRow;
+        }
+
+        @Ignore
+        @Test
+        public void getLastRowTest() {
+            assertArrayEquals(matrix.getRow(matrix.getRowsCount() - 1), lastRow);
+        }
+    }
+
+    @RunWith(Parameterized.class)
+    public static class GetFirstColumnTest {
+        private Matrix matrix;
+        private int[] firstColumn;
+
+        @Parameterized.Parameters(name = "{index}: checking first column of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0}},
+                    {"0\\n1", new int[] {0, 1}},
+                    {"0 1\\n2 3", new int[] {0, 2}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {0, 3, 6}}
```

Comment:
I wonder if it might be possible to format the matrices here in a more intuitive way? Something like the following (last row example):

```java
@Parameterized.Parameters(name = "{index}: checking first row of matrix constructed with string \"{0}\".")
public static Collection<Object[]> data() {
    return Arrays.asList(new Object[][]{
            {"0", new int[] {0}},
            {"0 1", new int[] {0, 1}},
            {"0\n1", new int[] {0}},
            {"0 1\n2 3", new int[] {0, 1}},
            {  "0 1 2\n"
             + "3 4 5\n"
             + "6 7 8", new int[] {0, 1, 2}}
    });
}
```

I find it a little easier to draw the connection between the input and the expected output with that formatting, but others might disagree...?